### PR TITLE
ZCS-11437  Add new CLI to manage global S3 buckets

### PR DESCRIPTION
**Pull request by inpati**
**Problem Statement**
Create CLI utility to provide CLI interface for S3ConfigBucket Soap requests

**Solution**
Created new class and file named "**VolumeS3ConfigCLI**" which is CLI interface for s3BucketConfig requests

### CODE REVIEW

File: store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+    // variables
+    private ZAuthToken auth;
+    private String storeProvider = "";
+    private String bucketId = "";
+    private String bucketName = "";
+    private String accessKey = "";;
+    private String secretKey = "";
+    private String destPath = "";
+    private String region = "";
+    private String account = "";
```
Comment (RakeshAMore) [inline store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java:80]: You don't need following parameters for S3 
 private String account = "";
  private String namespace = "";
    private String proxyPort = "";
    private String accountPort = "";
    private String url = "";

File: store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+        // Options
+        Options options = getOptions();
+        options.addOptionGroup(og);
+        options.addOption(O_BN, "bucketName", true, "Specify name for s3BucketConfig");
+        options.addOption(O_AK, "accessKey", true, "Specify access key for s3BucketConfig");
+        options.addOption(O_SK, "secretKey", true, "Specify secret key for s3BucketConfig");
+        options.addOption(O_DP, "destPath", true, "Specify destination path for s3BucketConfig");
+        options.addOption(O_R, "region", true, "Specify region for s3BucketConfig");
+        options.addOption(O_URL, "url", true, "Specify url for s3BucketConfig");
+        options.addOption(O_SP, "type", true, "Specify storeProvider to add s3BucketConfig");
```
Comment (RakeshAMore) [inline store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java:109]: it should be storageProvider right?

File: store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+     * @throws IOException, ServiceException
+     */
+    private void printS3BucketConfig(GlobalS3BucketConfiguration item) {
+        System.out.println("  globalBucketUUID: " + item.getGlobalBucketUUID());
+        System.out.println("        bucketName: " + item.getBucketName());
+        System.out.println("     storeProvider: " + item.getStoreProvider());
+        System.out.println("          protocol: " + item.getProtocol());
+        System.out.println("         storeType: " + item.getProtocol());
+        System.out.println("         accessKey: " + item.getAccessKey());
+        System.out.println("         secretKey: " + item.getSecretKey());
```
Comment (RakeshAMore) [inline store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java:192]: remove not required parameters.

**Review by RakeshAMore — CHANGES_REQUESTED** [review]

File: store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java (hunk 1 above)
Comment (inpati) [reply store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java:80]: Removed unnecessary params

File: store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java (hunk 2 above)
Comment (inpati) [reply store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java:109]: should we keep storeProvider or storageProvider everywhere ?

File: store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java (hunk 3 above)
Comment (inpati) [reply store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java:192]: Removed

**Comment by inpati** [comment]
This PR is deprecated raised new PR --> https://github.com/Zimbra/zm-hsm-store/pull/13

**Pull request closed without merging**